lib/Importer/ONNXIFIModelLoader.cpp

```diff
@@ -46,20 +46,27 @@ static llvm::Error setTensorType(const ONNX_NAMESPACE::TypeProto &in,
   }
 }
 
-llvm::Error ONNXIFIModelLoader::loadInputs(ONNX_NAMESPACE::GraphProto &net) {
+llvm::Error ONNXIFIModelLoader::loadInputs(ONNX_NAMESPACE::GraphProto &net,
+                                           bool loadInputsAsPlaceholders) {
   for (const auto &in : net.input()) {
     // Skip static weights.
     if (tensors_.count(in.name())) {
       continue;
     }
 
-    Tensor T;
-    RETURN_IF_ERR(setTensorType(in.type(), &T));
-
-    Placeholder *placeholder;
-    ASSIGN_VALUE_OR_RETURN_ERR(
-        placeholder, createAndRegisterPlaceholder(in.name(), &T.getType()));
-    onnxNameToInputVars_.try_emplace(in.name(), placeholder);
+    if (loadInputsAsPlaceholders) {
+      Tensor T;
+      RETURN_IF_ERR(setTensorType(in.type(), &T));
+
+      Placeholder *placeholder;
+      ASSIGN_VALUE_OR_RETURN_ERR(
+          placeholder, createAndRegisterPlaceholder(in.name(), &T.getType()));
+      onnxNameToInputVars_.try_emplace(in.name(), placeholder);
+    } else {
+      std::unique_ptr<Tensor> T(new Tensor());
```

Comment:
We don't have `make_unique` and stuff like that?

Reply:
Idk, @rdzhabarov mentioned on an earlier PR of his that we shouldn't assume we have it.

Reply:
Hm, I thought we used gcc-5 in Travis, and make_shared has been around since 4.9.  But I'm not complaining because as long as we don't use make shared I can keep limping along with gcc 4.8 ;-)

Reply:
`make_shared` is c++11 while `make_unique` is c++14 IIRC.....

Reply:
Sorry, I meant make_unique.  I think make_shared is fine.

Reply:
There is also `llvm:: make_unique`